pkg/applicationserver/io/packages/alcsync/v1/package.go

```diff
@@ -73,9 +75,27 @@ func (a *alcsyncpkg) HandleUp(
 		return nil
 	}
 
-	commands, err := MakeCommands(msg, fPort, data)
+	eventBuilders := make(events.Builders, 0)
+
+	defer func(ids *ttnpb.EndDeviceIdentifiers) {
+		if err != nil {
+			eventBuilders = append(eventBuilders, EvtPkgFail.With(
+				events.WithIdentifiers(ids),
+				events.WithData(err),
+			))
+		}
+		publishEvents(ctx, eventBuilders...)
+	}(up.GetEndDeviceIds())
+
+	commands, evts, err := MakeCommands(msg, fPort, data)
 	if err != nil {
 		logger.WithError(err).Debug("Failed to parse frame payload into commands")
+	}
+
+	eventBuilders = append(eventBuilders, evts...)
+
+	if err != nil && len(commands) == 0 {
```

Comment:
I don't think this is an error, and if you return early you don't need to check `err`.